go/vt/autoschema/validator.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package autoschema
+
+import (
+	"errors"
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// Error: Table does not exist
+var ErrUnknownTable = errors.New("Table does not exist")
+
+// Error: Too many rows
+var ErrTooManyRows = errors.New("Too many rows")
+
+// Error: Invalid Sql Statement
+var ErrInvalidSql = errors.New("Invalid Sql Statement")
+
+// Error: In Transaction at the end of sql statements
+var ErrInTransactionAtTheEnd = errors.New("In Transaction at the end of sql statements")
+
+// Validator apply a set of rules to perform on a schema change
+type Validator struct {
+}
+
+// TableSchema provides a view of MySQL table schema
+type TableSchema struct {
+	numRows int
+	columns []string
+}
+
+// Context carries any necessary information to help Validator apply rules
+type Context struct {
+	isInTransaction bool
+	maxAlterRows    int
+	tables          map[string]*TableSchema
+}
+
+// NewValidator creates a new Validator instance with given list of rules
+func NewValidator() (*Validator, error) {
+	return &Validator{}, nil
+}
+
+// Validate a list of sql string
+func (v *Validator) Validate(sqlChan <-chan string, ctx *Context) error {
+	for sql := range sqlChan {
+		// TODO: add Transaction statement support
+		ast, err := sqlparser.Parse(sql)
+		if err != nil {
+			return ErrInvalidSql
+		}
+		switch ast.(type) {
+		case *sqlparser.DDL:
+			validateAlter(ast.(*sqlparser.DDL), ctx)
+		}
+	}
+	if ctx.isInTransaction {
+		return ErrInTransactionAtTheEnd
+	}
+	return nil
+}
+
+// Validate alter statement
```

Comment:
Comment is redundant. You can instead write something like validateAlter checks whether...